tests/kernel/protection/src/main.c

```diff
@@ -48,11 +48,17 @@ void _SysFatalErrorHandler(unsigned int reason, const NANO_ESF *pEsf)
 #define DO_BARRIERS() do { } while (0)
 #endif
 
+/* i386 MMU doesn't control execute capabilities, only on x86_64 */
+#ifndef CONFIG_X86
+#define NO_EXECUTE_SUPPORT 1
+#endif
```

Comment:
Whine: seems like the proper magic would be to enumerate "NX_SUPPORT" or whatever as a predicate macro instead of assuming that every architecture other than i386 supports it.